### PR TITLE
Hub Interfaces

### PR DESCRIPTION
**Pull request by CalebCourier**
There are two main parts to this PR.

1. The actual schemas.  These represent our data structures for Validator Manifests and will be consumed in the various hub codebases: [validator-hub-service](https://github.com/guardrails-ai/validator-hub-service) and [validator-hub-ui](https://github.com/guardrails-ai/validator-hub-ui)
2. The demo code for how we can use the above schemas.  There is one path for bundling the schemas together and generating some Typescript types.  There are multiple paths for trying to get useful Python classes from the schemas; each with varying levels of success.  The `openapi-python-client`, which we also use for the [guardrails-api-client](https://github.com/guardrails-ai/guardrails-api-client), is the most useful path shown.

One caveat that I don't like but will have to live with for now is the `Manifest.parameters` type.  It can and should b JsonSchema's meta schema, but no tool in the Python toolchain can handle self-referencing schemas.  Since we likely won't need the full flexibility of the entire meta schema, I included an abridged version for now.

### CODE REVIEW

File: schemas/hub/manifest.json [hunk 1]
```diff
@@ -0,0 +1,116 @@
+{
+  "$id": "https://raw.githubusercontent.com/guardrails-ai/interfaces/schemas/hub/manifest.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Manifest",
+  "type": "object",
+  "properties": {
```
Comment (zsimjee) [inline schemas/hub/manifest.json:6]: seems like a lot of properties in one place. What do you think of (later) splitting this up into the following categories?
1. github properties
2. hub-entry properties (submitter/id)
3. runtime properties

File: schemas/hub/tags.json [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    },
+    "infrastructureRequirements": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "enum": [
+          "ML",
+          "NA",
+          "LLM",
+          "rule"
```
Comment (zsimjee) [inline schemas/hub/tags.json:51]: Rule?

**Review by zsimjee — APPROVED** [review]

File: schemas/hub/tags.json (hunk 2 above)
Comment (CalebCourier) [reply schemas/hub/tags.json:51]: All of the tag values are taken from what is actually in the manifests in [the hub](https://github.com/guardrails-ai/guardrails-hub).  If there's a value we don't like we can remove it, this was just a starting point.

**Review by zsimjee — APPROVED** [review]

File: schemas/hub/tags.json (hunk 2 above)
Comment (zsimjee) [reply schemas/hub/tags.json:51]: oh I meant "Rule" instead of "rule" just to match

File: schemas/hub/manifest.json (hunk 1 above)
Comment (zsimjee) [reply schemas/hub/manifest.json:6]: yeah namespacing

**Pull request merged**